### ISSUE TITLE
Build homogeneity for CMake options?

### ISSUE THREAD

**Issue by marcodelapierre** [ROCm/hipFFT#6]
Both hipBLAS and hipSOLVER use "USE_CUDA=ON" to specify a CUDA backend build,
whereas hipFFT currently uses "BUILD_WITH_LIB=CUDA".
(hipSPARSE uses yet another one)

It would probably ease installation efforts if the syntax could be made more homogeneous.

Thanks,

**Comment by malcolmroberts** [comment]
Thanks for raising this issue.  We're discussing how best to solve this - one advantage of  "BUILD_WITH_LIB=CUDA" is that we can specify other libs, besides ROCM and CUDA, of we decide to support these in the future.  But having the same syntax between the libraries is definitely a good idea.

**Comment by marcodelapierre** [comment]
I agree with the larger flexibility allowed by your current syntax..so perhaps that's the one to be used across all HIP numerical libraries? Probably a good conversation with teams working on the other ones (hipBLAS, SOLVER, SPARSE, ..)

**Comment by malcolmroberts** [comment]
Absolutely.  I have brought this up with the team, and I will see what we can agree on.

**Comment by ppanchad-amd** [comment]
There is an internal ticket tracking this feature request. Thanks!

**Comment by ammallya** [comment]
Imported to ROCm/rocm-libraries
